src/reference/antora/modules/ROOT/pages/mail.adoc

```diff
@@ -288,16 +506,98 @@ The following example assumes the payload is a Java bean with appropriate getter
 	<int-mail:subject expression="payload.subject" overwrite="true"/>
 </int-mail:header-enricher>
 ----
+======
+
 
 Alternatively, you can use the `value` attribute to specify a literal.
 You also can specify `default-overwrite` and individual `overwrite` attributes to control the behavior with existing headers.
 
+[[configuring-inbound-channel-adapters]]
+== Configuring Inbound Channel Adapters
+
 To configure an inbound channel adapter, you have the choice between polling or event-driven (assuming your mail server supports IMAP `idle` -- if not, then polling is the only option).
 A polling channel adapter requires the store URI and the channel to which to send inbound messages.
 The URI may begin with `pop3` or `imap`.
 The following example uses an `imap` URI:
 
-[source,xml]
+[tabs]
+======
+Java::
++
+[source,java,role="primary"]
+----
+@Bean
+@InboundChannelAdapter(value = "receiveChannel", poller = @Poller(fixedDelay = "5000"))
+public MailReceivingMessageSource mailMessageSource(ImapMailReceiver imapMailReceiver) {
+    return new MailReceivingMessageSource(imapMailReceiver);
+}
+
+@Bean
+public ImapMailReceiver imapMailReceiver(Properties javaMailProperties) {
+    ImapMailReceiver receiver = new ImapMailReceiver("imaps://[username]:[password]@imap.gmail.com/INBOX");
+    receiver.setShouldDeleteMessages(true);
+    receiver.setShouldMarkMessagesAsRead(true);
+    receiver.setMaxFetchSize(1);
+    receiver.setJavaMailProperties(javaMailProperties);
+
+    return receiver;
+}
+
+----
+Java DSL::
++
+[source,java,role="secondary"]
+----
+@Bean
+public IntegrationFlow imapMailInboundFlow(Properties javaMailProperties, MessageChannel receiveChannel) {
+    return IntegrationFlow
+        .from(Mail.imapInboundAdapter("imaps://[username]:[password]@imap.gmail.com/INBOX")
+                .shouldDeleteMessages(true)
+                .shouldMarkMessagesAsRead(true)
+                .javaMailProperties(javaMailProperties)
+                .maxFetchSize(1),
+            e -> e.poller(Pollers.fixedRate(5000)))
+        .channel(receiveChannel)
+        .get();
+}
+----
+Kotlin DSL::
++
+[source,kotlin,role="secondary"]
+----
+@Bean
+fun imapMailInboundFlow(javaMailProperties: Properties, receiveChannel: MessageChannel): IntegrationFlow {
```

Comment:
I think the return type is not necessary here.
Just ` = integrationFlow() {...}`
And that `return` word, too.